index.js

```diff
@@ -155,179 +132,110 @@ function readProtocolData (reader) {
-  const amount =twoNumbersToString(reader.readUInt64())
-  const executionCondition = base64url(reader.read(32))
-  const expiresAt = readGeneralizedTime(reader)
-  const protocolData = readProtocolData(reader)
-
-  return { requestId, transferId, amount, executionCondition, expiresAt, protocolData }
-}
-
-function serializeFulfill ({ transferId, fulfillment }, requestId, protocolData) {
-  const transferIdBuffer = Buffer.from(transferId.replace(/-/g, ''), 'hex')
-  const fulfillmentBuffer = Buffer.from(fulfillment, 'base64')
-  const writer = new Writer()
-
-  writer.write(transferIdBuffer)
-  writer.write(fulfillmentBuffer)
-  writeProtocolData(writer, protocolData)
-
-  return writeEnvelope(TYPE_FULFILL, requestId, writer.getBuffer())
-}
-
-function deserializeFulfill (buffer) {
-  const { requestId, data } = readEnvelope(buffer)
-  const reader = new Reader(data)
-
-  const transferId = uuidParse.unparse(reader.read(16))
-  const fulfillment = base64url(reader.read(32))
-  const protocolData = readProtocolData(reader)
-
-  return { requestId, transferId, fulfillment, protocolData }
-}
-
-function serializeReject ({ transferId, rejectionReason }, requestId, protocolData) {
-  const transferIdBuffer = Buffer.from(transferId.replace(/-/g, ''), 'hex')
-  const rejectionReasonBuffer = maybeSerializeIlpError(rejectionReason)
-
-  const writer = new Writer()
-  writer.write(transferIdBuffer)
-  writer.write(rejectionReasonBuffer)
-  writeProtocolData(writer, protocolData)
-
-  return writeEnvelope(TYPE_REJECT, requestId, writer.getBuffer())
-}
-
-function deserializeReject (buffer) {
-  const { requestId, data } = readEnvelope(buffer)
-  const reader = new Reader(data)
-
-  const transferId = uuidParse.unparse(reader.read(16))
-  const rejectionReason = readIlpError(reader)
-  const protocolData = readProtocolData(reader)
-
-  return { requestId, transferId, rejectionReason, protocolData }
-}
-
-function serializeMessage (requestId, protocolData) {
+function serialize (obj) {
   const writer = new Writer()
-
-  writeProtocolData(writer, protocolData)
-
-  return writeEnvelope(TYPE_MESSAGE, requestId, writer.getBuffer())
-}
-
-function deserializeMessage (buffer) {
-  const { requestId, data } = readEnvelope(buffer)
-  const reader = new Reader(data)
-
-  const protocolData = readProtocolData(reader)
-  return { requestId, protocolData }
-}
-
-function deserializeClpPacket (buffer) {
-  const {typeString, packet} = _toTypeString(buffer)
-  return {
-    type: buffer[0],
-    typeString,
-    packet
+  switch (obj.type) {
+  case TYPE_ACK:
+  case TYPE_RESPONSE:
+  case TYPE_MESSAGE:
+    writeProtocolData(writer, obj.data.protocolData)
+  break
+
+  case TYPE_ERROR: {
+    const ilpPacket = maybeSerializeIlpError(obj.data.rejectionReason)
+    writer.write(ilpPacket)
+    writeProtocolData(writer, obj.data.protocolData)
+  } break
+
+  case TYPE_PREPARE: {
+    const transferIdBuffer = Buffer.from(obj.data.transferId.replace(/\-/g, ''), 'hex')
```

Comment:
That's a zombie bug. I removed the backslash in the previous commit and now its back ;-) (that backslash is unnecessary)